dask_ml/wrappers.py

```diff
@@ -662,9 +663,28 @@ def _get_output_dask_ar_meta_for_estimator(model_fn, estimator, input_dask_ar):
     """
     # sklearn fails if input array has size size
     # It requires at least 1 sample to run successfully
-    ar = np.zeros(
-        shape=(1, input_dask_ar.shape[1]),
-        dtype=input_dask_ar.dtype,
-        like=input_dask_ar._meta,
-    )
+    imput_meta = input_dask_ar._meta
+    if hasattr(imput_meta, "__array_function__"):
+        ar = np.zeros(
+            shape=(1, input_dask_ar.shape[1]),
+            dtype=input_dask_ar.dtype,
+            like=imput_meta,
+        )
+    elif "scipy.sparse" in type(imput_meta).__module__:
+        # sparse matrices dont support
+        # `like` due to non implimented __array_function__
+        # Refer https:/q/github.com/scipy/scipy/issues/10362
+        # Note below works for both cupy and scipy sparse matrices
+        ar = type(imput_meta)((1, input_dask_ar.shape[1]), dtype=input_dask_ar.dtype)
+    else:
+        msg = (
+            "\nYou did not provide metadata, so Dask is running the"
```

Comment:
```suggestion
            "\nYou did not provide metadata for the output, so Dask is running the"
```

I personally got slightly confused about which metadata was being talked about here: `input_dask_ar._meta` or metadata passed in with `predict_meta`/`predict_proba_meta`/`transform_meta`. I think this will only be confusing for developers who have read the code above this bit, since users will just see the message. In any case, it might help to make the message a bit clearer.

Reply:
Reworded it to make as clear as i could. 

This is how it will look the user now. 

```python
UserWarning: Metadata for predict_proba is not provided, so Dask is running the predict_proba function on a small dataset to guess output metadata. As a result, It is possible that Dask will guess incorrectly.
To silence this warning, provide explicit `predict_proba_meta` to the dask_ml.wrapper.
Example: 
wrap_clf = dask_ml.wrappers.Incremental(GradientBoostingClassifier(), predict_proba_meta = np.array([1],dtype=np.int8))
  warnings.warn(msg)
```

Thanks for the inputs, please  feel free to let me know if it is still confusing.